internal/ad/krb5_test.go

```diff
@@ -65,6 +66,10 @@ func TestTicketPath(t *testing.T) {
 				require.Error(t, err, "TicketPath should have errored out")
 				return
 			}
+			if tc.wantErrType != nil {
```

Comment:
Just a nitpick: I think that could be only with wantErr, or maybe you want to if tc.wantErr || tc.wantErrType != " nil, then you check for an error, and readd a wantErrType check nested.

I’m afraid that we set `wantErr: true` and see the test passing (we return an error afterall), but then, don’t check the type because of the previous early return.